src/main/java/org/apache/commons/io/IOUtils.java

```diff
@@ -2664,6 +2664,50 @@ public static byte[] toByteArray(final InputStream input, final int size) throws
         return data;
     }
 
+
+    /**
+     *
+     * Gets the contents of an {@link InputStream} as a {@code byte[]}.Use this method when you want to specify the
+     * indexes that you want to read from the InputStream.
+     *
+     * @param input
+     * @param offset The offset from the start of the InputStream to start ready - index based
+     * @param length The end of the InputStream to stop reading - index based
+     * @return byte[]
+     * @throws IOException if an I/O error occurs or {@link InputStream} length is smaller than parameter {@code size}.
+     * @throws NullPointerException if offset is greater than length or array index is out of bounds
+     * @throws IllegalArgumentException if {@code size} is less than zero.
+     * @since 2.12.0
+     */
+    public static byte[] toByteArray(final InputStream input, final int offset, final int length) throws IOException {
+
+        if (length < 0) {
+            throw new IllegalArgumentException("Size must be equal or greater than zero: " + length);
+        }
+        if (offset < 0 ) {
+            throw new IllegalArgumentException("Offset must be grater than or equal to zero: " + offset);
+        }
+        if (length < offset) {
```

Comment:
The index and length are unrelated, the length is the length of the desired result. The question is what should happen if there is not enough file data.